Remove `Element::nodeNamePreservingCase`

Ahmad-S792 opened the pull request:
#### 4d2a8e2a2a0264d9dc994ddd32d7da3050736e8c
<pre>
Remove `Element::nodeNamePreservingCase`

<a href="https://bugs.webkit.org/show_bug.cgi?id=266600">https://bugs.webkit.org/show_bug.cgi?id=266600</a>

Reviewed by Ryosuke Niwa.

Merge: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=166250

There is no need for this method, we can just use tagQName and it is just simplification.

* Source/WebCore/dom/Element.cpp:
(Element::nodeNamePreservingCase): Deleted
* Source/WebCore/dom/Element.h: &apos;nodeNamePreservingCase&apos; definition deleted
* Source/WebCore/editing/MarkupAccumulator.cpp:
(MarkupAccumulator::appendEndTag):
(MarkupAccumulator::appendOpenTag):

Canonical link: <a href="https://commits.webkit.org/272256@main">https://commits.webkit.org/272256@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/038886327aabdc2b889aeb0f92340ecd31bb83ea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31100 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32788 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33606 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28079 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31863 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7030 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27914 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31432 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8234 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27804 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7073 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7241 "Too many flaky failures: imported/w3c/web-platform-tests/css/css-cascade/scope-hover.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-partially-onscreen-new.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-partially-onscreen-old.html, imported/w3c/web-platform-tests/css/selectors/invalidation/user-valid-user-invalid.html, imported/w3c/web-platform-tests/css/selectors/user-invalid.html, imported/w3c/web-platform-tests/css/selectors/valid-invalid-form-fieldset.html, imported/w3c/web-platform-tests/html/semantics/disabled-elements/disabled-event-dispatch.tentative.html, imported/w3c/web-platform-tests/html/semantics/invokers/invoketarget-button-event-dispatch.tentative.html, imported/w3c/web-platform-tests/html/semantics/selectors/pseudo-classes/readwrite-readonly.html, imported/w3c/web-platform-tests/mixed-content/gen/worker-classic.http-rp/opt-in/fetch.https.html, imported/w3c/web-platform-tests/webrtc-extensions/RTCRtpSynchronizationSource-senderCaptureTimeOffset.html (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27694 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34945 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28304 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28156 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33377 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7282 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5338 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31212 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8973 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7325 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7985 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7820 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/editing/MarkupAccumulator.cpp:
```diff
@@ -365,11 +365,16 @@ void MarkupAccumulator::startAppendingNode(const Node& node, Namespaces* namespa
         m_nodes->append(const_cast<Node&>(node));
 }
 
+static String nodeNamePreservingCase(const Element& element)
```
I don't think this helper is necessary. We can just repeat `element.tagQName.toString()`.

Ahmad-S792 replied:
@rniwa - Done. Thanks for suggestion.

Ahmad-S792 committed:
Remove `Element::nodeNamePreservingCase`

https://bugs.webkit.org/show_bug.cgi?id=266600

Reviewed by Ryosuke Niwa.

Merge: https://src.chromium.org/viewvc/blink?view=revision&revision=166250

There is no need for this method, we can just use tagQName and it is just simplification.

* Source/WebCore/dom/Element.cpp:
(Element::nodeNamePreservingCase): Deleted
* Source/WebCore/dom/Element.h: 'nodeNamePreservingCase' definition deleted
* Source/WebCore/editing/MarkupAccumulator.cpp:
(MarkupAccumulator::appendEndTag):
(MarkupAccumulator::appendOpenTag):

Canonical link: https://commits.webkit.org/272256@main
